x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -96,6 +104,36 @@ public ConnectorIndexService(Client client) {
         this.clientWithOrigin = new OriginSettingClient(client, CONNECTORS_ORIGIN);
     }
 
+    /**
+     * Returns the {@link SystemIndexDescriptor} for the Connector system index.
+     *
+     * @return The {@link SystemIndexDescriptor} for the Connector system index.
+     */
+    public static SystemIndexDescriptor getSystemIndexDescriptor() {
+        PutIndexTemplateRequest request = new PutIndexTemplateRequest();
+        String templateSource = TemplateUtils.loadTemplate(
+            "/elastic-connectors.json",
+            Version.CURRENT.toString(),
+            CONNECTORS_MAPPING_VERSION_VARIABLE,
+            Map.of(CONNECTORS_MAPPING_MANAGED_VERSION_VARIABLE, Integer.toString(CONNECTORS_INDEX_VERSION))
+        );
+        request.source(templateSource, XContentType.JSON);
+
+        // The index pattern needs a stricter regex to prevent conflicts with .elastic-connectors-sync-jobs
+        return SystemIndexDescriptor.builder()
+            .setIndexPattern(CONNECTOR_INDEX_NAME + "-v*")
```

Comment:
Nitpick: It would be good to define the full names we're using as variables, so we don't hard code partial strings when building the system index. 

It's not necessary for this PR, but if you ever need to make changes to the system index schema you'll want to manage versions using a version ID. (`QueryRulesIndexService` has an example of how to do this).

Reply:
I fixed the consts so they're a bit cleaner.

Regarding the version ID, I tried to do this as well as it seemed like a good idea, but I couldn't compile afterwards and couldn't figure out why, so I can do this in another PR.

Is this something we should have right away or just to keep in mind if we update the index schema?

Reply:
You're good with that until you need version 2 😉